### ISSUE TITLE
Don't add/patch core operators that rxjs-etc needs internally

### ISSUE THREAD

**Issue by jayphelps** [cartant/rxjs-etc#1]
Currently some of the custom operators in rxjs-etc import rxjs core operators by using the `rxjs/add` patching modules. This leaks implementation details and means it's now possible that someone may depend on those core operators existing on the Observable prototype, but not import them. If rxjs-etc would suddenly stop patching a particular core operator it would be a breaking change.

Instead, you can import the operators directly and using `call`:

```ts
import { from } from "rxjs/observable/from"; 
import { concat } from "rxjs/operator/concat"; 

export function endWith(...args) { 
     let scheduler = args[args.length - 1];
     if (isScheduler(scheduler)) { 
         args.pop(); 
     } else { 
         scheduler = null; 
     } 
     return (source) => concat.call(source, from(args, scheduler)); 
 } 
```

I imagine you already realized this since it's apparent you know RxJS well. Mostly recording here to track. 😄  Ideally I would just do this for you cause it's relatively easy, but I don't have much OSS time right now (recovering from surgery still)

**Comment by cartant** [comment]
Thanks. I'm aware of it. And I agree with you.

I would prefer to use that approach, but what's stopping me is that `call` loses the type information. I would be much happier with it if `call` were to return something appropriate, rather than `any`.

The implementations in this package are relatively simple, so dealing with the lost type information is not that significant an issue. And they could be explicitly typed without too much fuss.

I've found that losing the type information has cost me a lot of time when dealing with longer observable chains in application-specific code, so I've favoured patching the prototype for that reason. Less of an issue here, though.

Now, that you've pointed it out, I'm inclined to change it, as it doesn't serve as a particularly good example of how to implement a library.

I've not yet looked at/listened to Ben Lesh's talk, from yesterday, re: changes in v6, but I understand there is some discussion of lettable operators and a secret prototype, which does sound good.

(And I wish you well with your recovery.)
